### PR TITLE
Url tab cleanups for result details

### PR DESCRIPTION
**Pull request by asdil12**
- Only redirect to running tab when no `#` is in the url
- Clear url hash on test step result preview close
- Test for url hash cleanup after preview close

### CODE REVIEW

File: assets/javascripts/test_result.js [hunk 1]
```diff
@@ -187,7 +188,9 @@ function setupResult(state, jobid) {
     if (state == "scheduled") {
       setResultHash("#settings", true);
     } else if (state == "running" || state == "waiting") {
-      setResultHash("#live", true);
+      if (!document.location.href.endsWith("#")) {
```
Comment (coolo) [inline assets/javascripts/test_result.js]: this is very confusing. There is a documentation.location.href and a window.location ? What is the difference?

**Comment by asdil12** [comment]
nothing

**Comment by coolo** [comment]
then please don't use document

**Comment by asdil12** [comment]
done

**Pull request merged**